core/kernel/tee_ta_manager.c

```diff
@@ -344,6 +344,11 @@ TEE_Result tee_ta_close_session(struct tee_ta_session *csess,
 	}
 
 	tee_ta_unlink_session(sess, open_sessions);
+#if defined(CFG_TA_GPROF_SUPPORT)
+       if (sess->sbuf)
+               free(sess->sbuf->samples);
```

Comment:
Good catch, I'll scratch that. I'm wondering why I got no error or crash however.